Mathlib/Algebra/DirectLimit.lean

```diff
@@ -690,6 +712,15 @@ theorem lift_unique [IsDirected ι (· ≤ ·)] (F : DirectLimit G f →+* P) (x
   · exact DirectLimit.induction_on x fun i x => by simp [lift_of]
 #align ring.direct_limit.lift_unique Ring.DirectLimit.lift_unique
 
+lemma lift_injective [Nonempty ι] [IsDirected ι (· ≤ ·)]
```

Comment:
Is there a reason this one needs nonempty but the others don't?

Reply:
The free module / abelian group on zero generators is the zero module / abelian group, so any map out of them are injective.
However the free commutative ring on zero generators is Z (in general it's MvPolynomial over Z), which maps to quotients Z/nZ non-injectively (when n is not 1).